inspirehep/modules/records/validators/rules.py

```diff
@@ -0,0 +1,136 @@
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014, 2015, 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+'''Validator classes for Invenio Record Editor.'''
+
+from __future__ import absolute_import, division, print_function
+from .helpers import (
+    check_if_field_exists_in_dict_list_x_times,
+    check_field_values_not_in_required_values_for_record,
+    FIELD_REQUIRE_FIELD_VALUES_TEMPLATE,
+    check_if_listfield_property_values_are_valid,
+    query_db_for_duplicates_for_field_listitem_with_index,
+    FIELD_VALIDATION_TEMPLATE
+)
+from jsonschema import ValidationError
+from idutils import is_isbn
+import pycountry
+from collections import deque
+
+
+'''
+Error Validator Functions
+'''
+
+
+def check_for_author_or_corporate_author_to_exist(record):
+    if all(k not in record for k in ['authors', 'corporate_author']):
+        yield ValidationError(path=deque(['']),
```

Comment:
It seems like `jsonschemas` will create the `deque` for you - https://github.com/Julian/jsonschema/blob/master/jsonschema/exceptions.py#L43 so no need to create it here.